sky/clouds/service_catalog/common.py

```diff
@@ -325,3 +324,15 @@ def accelerator_in_region_or_zone_impl(
         return _accelerator_in_region(df, accelerator_name, acc_count, region)
     else:
         return _accelerator_in_zone(df, accelerator_name, acc_count, zone)
+
+
+# Images
+def get_image_id_from_tag_impl(df: pd.DataFrame, tag: str,
+                               region: Optional[str]) -> Optional[str]:
+    """Returns the image ID for the given tag and region."""
+    df = df[df['Tag'] == tag]
+    if region is not None:
+        df = df[df['Region'] == region]
+    assert len(
```

Comment:
Can't we hit this assert when region is None? E.g., many regions offer the tag `sky:k80-ubuntu-1804`?

Reply:
I'm missing something: what happens if we have
```
resources:
  cloud: aws
  image_id: sky:gpu-ubuntu-18.04
```
which will find multiple images with this tag, hitting this assert?

Reply:
It won't hit the assertion, since the method will only be called after the failover mechanism decides the region.